python/Scripts/baketextures.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python
+'''
+Generate the baked mtlx file and baked textures programmatically.
+'''
+
+import sys, os, string
+import glfw
+import struct
+import MaterialX as mx
+from MaterialX import PyMaterialXRender as r
+from MaterialX import PyMaterialXRenderGlsl as rg
+from MaterialX import PyMaterialXGenShader as gs
+
+def usage():
+    print("baketextures.py: bake out MaterialX shaderref inputs into textures.")
+    print("Usage:  baketextures.py [options] <file.mtlx>" )
+    print("    -h[elp]         Print usage information")
+    print("    -hdr            Save images to hdr format")
+    print("    -t[exRes] <res> Resolution at which to save textures")
+
+def main():
+    if len(sys.argv) < 2:
+        usage()
+        sys.exit(0)
+
+    hdr = False
+    texres = 1024
+    for i in range(1, len(sys.argv)):
+        arg = sys.argv[i]
+        if arg in ['-h', '-help', '--help']:
+            usage()
+            sys.exit(0)
+        elif arg in ['-hdr', '-HDR']:
+            hdr = True
+        elif (arg in ['-t', 'texRes', 'texres'] and i + 1 < len(sys.argv)):
+            texres = int(sys.argv[i + 1])
+            i += 1
+        else:
+            filename = arg
```

Comment:
We should use the built-in `argparse` module for this. I believe for this inputs, that's would be close to this (though you don't need all my comments):

```
import argparse

parser = argparse.ArgumentParser(description="Bake out MaterialX sharderref inputs into textures.")

# Here's a boolean flag
parser.add_argument("--hdr", "--HDR", dest="hdr", type="store_true", help="Save images to hdr format.")

# This one takes the user input and converts it to an int, with a default value.
# `dest` in optional (default to flag name), but good to be explicit when you have multiple flag names
parser.add_argument("-t", "--texRes", "--texres", dest="tex_res", type=int, default=1024, help="Resolution at which to save textures.")

# If `parse_args` is called with no input, it uses `sys.argv`. If a help flag is passed, this automatically generates and prints usage
opts = parser.parse_args()

# Values are stored as attributes on this object.
# opts.hdr
# opts.tex_res
```
https://pymotw.com/3/argparse/

Reply:
Amazing thank you!